LykkeApi2/Controllers/WhitelistingsController.cs

```diff
@@ -21,87 +29,211 @@ namespace LykkeApi2.Controllers
                 throw LykkeApiErrorException.BadRequest(LykkeApiErrorCodes.Service.AssetUnavailable);
+
+            var wallets = await _clientAccountService.Wallets.GetClientWalletsFilteredAsync(_requestContext.ClientId, owner: OwnerType.Spot, walletType: WalletType.Trusted);
+
+            if(wallets.All(x => x.Id != request.WalletId))
+                throw LykkeApiErrorException.BadRequest(LykkeApiErrorCodes.Service.InvalidInput);
             
-            if(request.Code2Fa == "2222")
+            var siriusAssetResponse = await _siriusApiClient.Assets.SearchAsync(new AssetSearchRequest {Id = asset.SiriusAssetId});
+            var siriusAsset = siriusAssetResponse.Body.Items.Single();
+            
+            var siriusAccount = await TryGetSiriusAccountAsync(_requestContext.ClientId, request.WalletId);
+            
+            if(siriusAccount == null)
                 throw LykkeApiErrorException.BadRequest(LykkeApiErrorCodes.Service.BlockchainWalletDepositAddressNotGenerated);
+
+            var whitelistedItems = await GetWhitelistItemsAsync(siriusAccount.Id);
             
-            if(request.Code2Fa == "3333")
+            if(whitelistedItems.Any(x => x.Details.Address == request.AddressBase && x.Details.Tag == request.AddressExtension))
                 throw LykkeApiErrorException.BadRequest(LykkeApiErrorCodes.Service.AddressAlreadyWhitelisted);
             
-            var check2FaResult = await _google2FaService.Check2FaAsync<string>(_requestContext.ClientId, request.Code2Fa);
+            var requestId = $"{request.WalletId}:{request.Name}:{request.AssetId}:{request.AddressBase}:{request.AddressExtension ?? string.Empty}";
 
-            if (check2FaResult != null)
-                return Ok(check2FaResult);
+            var result = await _siriusApiClient.WhitelistItems.CreateAsync(new WhitelistItemCreateRequest
+            {
+                Name = request.Name,
+                Scope = new WhitelistItemScopeModel
+                {
+                    BrokerAccountId = _siriusApiServiceClientSettings.BrokerAccountId,
+                    AccountId = siriusAccount.Id,
+                    AccountReferenceId = request.WalletId
+                },
+                Details = new WhitelistItemDetailsModel
+                {
+                    AssetId = siriusAsset.Id,
+                    BlockchainId = siriusAsset.BlockchainId,
+                    Address = request.AddressBase,
+                    Tag = request.AddressExtension,
+                    TagType = string.IsNullOrWhiteSpace(request.AddressExtension)
+                        ? new NullableWhitelistItemTagModel
+                        {
+                            Null = NullValue.NullValue
+                        }
+                        : new NullableWhitelistItemTagModel
+                        {
+                            TagType = int.TryParse(request.AddressExtension, out _) ? WhitelistItemTagModel.Number : WhitelistItemTagModel.Text
```

Comment:
it depends on the blockchain actually. There is no such meta-information in LW integrations. Let's just create a method to get `WhitelistItemTagModel` depending on the `asset.BlockchainIntegrationLayerType` with the switch with the only `default` case which always throws `IndexOutOfRangeException`. Will add particular cases on demand